lib/mindwendel_web/live/idea_live/form_component.ex

```diff
@@ -48,12 +48,15 @@ defmodule MindwendelWeb.IdeaLive.FormComponent do
   end
 
   defp save_idea(socket, :new, idea_params) do
-    Mindwendel.Accounts.update_user(socket.assigns.current_user, %{
-      username: idea_params["username"]
-    })
+    {:ok, user} =
+      Mindwendel.Accounts.update_user(socket.assigns.current_user, %{
+        username: idea_params["username"]
+      })
 
     case Ideas.create_idea(Map.put(idea_params, "user_id", socket.assigns.current_user.id)) do
       {:ok, _idea} ->
+        send(self(), {:user_updated, user})
```

Comment:
shouldn't `:user_updated` be issued earlier after the user was updated vs. just when the idea is successfully updated?

Reply:
well imagine you type in an idea and there is an error. Now close the modal, open a new one and your username has changed? Personally, I only expect data to change if there was valid feedback and not if errors were present. Thats why I put it there.

Reply:
@JannikStreek but the user is already updated in the database at this point in time - so it's changed in the database but not in the UI which leads to inconsistencies. If that's the logic we go for, then the above user name change should only happen after the idea changed successfully? And both should run in a transaction as it seems we want both to fail or both to succeed? :eyes:

Reply:
good point, it should be consistent. I will double check that.

Reply:
also interesting is that the username is not saved when editing an idea - only when adding a new one :-D but thats a new issue